backend/dataall/base/aws/cognito.py

```diff
@@ -64,3 +63,6 @@ def list_cognito_groups(envname: str, region: str):
             )
             raise e
         return groups
+
+    def get_groups_for_user(self, user_id):
```

Comment:
To make the created ServiceProvider abstraction meaningful for this function as well, how about moving the logic relevant to this function [from the api handler](https://github.com/awslabs/aws-dataall/pull/897/files#diff-d9614c6d1ad54c1883eb1a16f3f4e342f93939f7b02ca8529c60dc0e2a85c98aR66) to here? You could omit the user_id input from the base class to make it work

Reply:
Currently, this function is not used when using the cognito stack. Thus for now, this function just returns nothing. The function that you were aluding to - [in the api handler](https://github.com/awslabs/aws-dataall/pull/897/files#diff-d9614c6d1ad54c1883eb1a16f3f4e342f93939f7b02ca8529c60dc0e2a85c98aR66) , would be getting groups information from the claims provided by the Cognito Authorizer. I am not sure if sections of this function would be a good fit here. Please let me know if the explanation I provided answers your question . Maybe I am not getting question 😅

Reply:
> The function that you were aluding to - [in the api handler](https://github.com/awslabs/aws-dataall/pull/897/files#diff-d9614c6d1ad54c1883eb1a16f3f4e342f93939f7b02ca8529c60dc0e2a85c98aR66) , would be getting groups information from the claims provided by the Cognito Authorizer.

But isn't it exactly what this function is supposed to do? To get the groups for the user - in case of Cognito auth by extracting them from the claims - in case of external Idp via some other custom implementation.

This change has no functional values, but since you created an abstraction via the Service provider factory, I think it would be cleaner to make use of its abstract methods.

Reply:
yes exactly. but as these are no functional changes it could also be in a followup not to block the merging

Reply:
Yes I think we can implement the function at a later time if that's not needed right now. Commented here. https://github.com/awslabs/aws-dataall/issues/899